### PR TITLE
Allow for different builders in cross-tests

### PR DESCRIPTION
**Pull request by QuentinI**
Closes #0000

### This PR: 
Adds an optional argument to cross-tests macro to allow specifying builder implementation(s)

### This PR does not: 
Change builder implementation in existing cross-tests

### Key places to review: 
Everything

### CODE REVIEW

**Review by bfish713 — APPROVED** [review]
I like this change, cc: @ss-es to double check the macro stuff

File: crates/macros/src/lib.rs [hunk 1]
```diff
@@ -19,6 +19,9 @@ use syn::{
 struct CrossTestData {
     /// imlementations
     impls: ExprArray,
+    /// builder impl
+    #[builder(default = "syn::parse_str(\"[SimpleBuilderImplementation]\").unwrap()")]
+    builder_impls: ExprArray,
```
Comment (bfish713) [inline crates/macros/src/lib.rs:24]: I assume this will eventually be used for some marketplace builder tests?

Comment (QuentinI) [reply crates/macros/src/lib.rs:24]: Yes, both here and in marketplace repos.

File: crates/macros/src/lib.rs [hunk 2]
```diff
@@ -19,6 +19,9 @@ use syn::{
 struct CrossTestData {
     /// imlementations
     impls: ExprArray,
+    /// builder impl
+    #[builder(default = "syn::parse_str(\"[SimpleBuilderImplementation]\").unwrap()")]
```
Comment (ss-es) [inline crates/macros/src/lib.rs:23]: I slightly prefer not having a default implementation (and just explicitly writing `SimpleBuilderImplementation` in all the integration tests), but I think it's fine either way and we could just change later if we want

**Review by ss-es — APPROVED** [review]
looks good to me!

**Pull request merged**